user/views.py

```diff
@@ -42,3 +40,30 @@ def post(self, request, *args, **kwargs):
             form.save()
             return redirect('user:shows')
         return render(request, self.template_name, {'form': form})
+
+@login_required
+def book_ticket(request, show_id):
+    show = Show.objects.get(id=show_id)
+    
+    if request.method == 'POST':
+        seat_number = request.POST.get('seat_number')
+
+        # Count tickets booked by the user this month
+        ticket_count = Ticket.objects.filter(
+            user=request.user,
+            booked_at__month=timezone.now().month,
+            booked_at__year=timezone.now().year
+        ).count()
+
+        if ticket_count >= 2:
+            return render(request, 'book_ticket.html', {'show': show, 'error': 'You can only book up to 2 tickets per month.'})
+
+        # Check if the seat is already booked
```

Comment:
on the frontend just display whatever seats are available instead of having user repeatedly select unavailable seats.